#VRML_SIM V7.4.3 utf8
WorldInfo {
  info [
    "The model of the Pioneer 2 robot"
    "Author: Simon Blanchoud"
    "Date: 02 SEP 2006"
  ]
  title "Pioneer 2"
}
Viewpoint {
  orientation 0.567834 -0.590381 -0.573598 4.21565
  position 0.110375 3.44553 0.00258298
}
Background {
  skyColor [
    0.4 0.7 1
  ]
}
DirectionalLight {
  ambientIntensity 0.2
  direction -1.41 -0.61 0.3
}
DirectionalLight {
  ambientIntensity 0.2
  direction 0.3 -0.7 0.5
  castShadows TRUE
}
DEF FLOOR Solid {
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0.460425 0.445021
        }
      }
      geometry DEF GROUND IndexedFaceSet {
        coord Coordinate {
          point [
            -1 0 -1
            1 0 -1
            1 0 1
            -1 0 1
          ]
        }
        coordIndex [
          3, 2, 1, 0, -1
        ]
      }
    }
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
      }
      geometry DEF WALLS IndexedFaceSet {
        coord Coordinate {
          point [
            -1 0 -1
            -1 0 1
            1 0 1
            1 0 -1
            -1 0.5 -1
            -1 0.5 1
            1 0.5 1
            1 0.5 -1
          ]
        }
        coordIndex [
          0, 4, 5, 1, 0, -1, 0, 3, 7, 4
          0, -1, 1, 5, 6, 2, 1, -1, 2, 6
          7, 3, 2, -1
        ]
      }
    }
  ]
  name "ground"
  boundingObject Group {
    children [
      USE WALLS
      USE GROUND
    ]
  }
}
DEF PIONEER2 DifferentialWheels {
  translation -0.67 0 -0.6
  rotation 0 1 0 -2.918
  children [
    TouchSensor {
      translation 8.48885e-07 0.12 0.250239
      rotation 0.976139 0.217096 -0.00458998 -1.52421e-10
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.6 0.4 0.2
            }
          }
          geometry DEF SENSOR_SHAPE Sphere {
            radius 0.07
          }
        }
      ]
      name "ts_back"
      boundingObject USE SENSOR_SHAPE
      physics DEF TOUCH_SENSOR_PHYSICS Physics {
        density -1
        mass 0.01
      }
    }
    TouchSensor {
      translation 0.160059 0.12 -0.0200283
      rotation 0.995742 0.0921762 -0.00127793 -3.73819e-10
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.6 0.4 0.2
            }
          }
          geometry USE SENSOR_SHAPE
        }
      ]
      name "ts_right"
      boundingObject USE SENSOR_SHAPE
      physics USE TOUCH_SENSOR_PHYSICS
    }
    TouchSensor {
      translation 0.100038 0.12 -0.100055
      rotation 0.446086 -0.893779 0.0465393 3.99284e-09
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.6 0.4 0.2
            }
          }
          geometry USE SENSOR_SHAPE
        }
      ]
      name "ts_front_right"
      boundingObject USE SENSOR_SHAPE
      physics USE TOUCH_SENSOR_PHYSICS
    }
    TouchSensor {
      translation -0.160058 0.12 -0.0200308
      rotation 0.996066 0.0885422 -0.00369364 -3.73376e-10
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.6 0.4 0.2
            }
          }
          geometry USE SENSOR_SHAPE
        }
      ]
      name "ts_left"
      boundingObject USE SENSOR_SHAPE
      physics USE TOUCH_SENSOR_PHYSICS
    }
    TouchSensor {
      translation -0.100036 0.12 -0.100059
      rotation 0.996066 0.0885427 -0.0036939 -3.73376e-10
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.6 0.4 0.2
            }
          }
          geometry USE SENSOR_SHAPE
        }
      ]
      name "ts_front_left"
      boundingObject USE SENSOR_SHAPE
      physics USE TOUCH_SENSOR_PHYSICS
    }
    Shape {
      appearance Appearance {
        material Material {
          ambientIntensity 0.34
          diffuseColor 1 0 0
        }
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.1 0.059 0.215
            0.1 0.059 0.215
            0.135 0.059 0.185
            0.135 0.059 -0.095
            0.08 0.059 -0.11
            -0.08 0.059 -0.11
            -0.135 0.059 -0.095
            -0.135 0.059 0.185
            -0.1 0.059 0.215
            -0.1 0.234 0.215
            0.1 0.234 0.215
            0.135 0.234 0.185
            0.135 0.234 -0.095
            0.08 0.234 -0.11
            -0.08 0.234 -0.11
            -0.135 0.234 -0.095
            -0.135 0.234 0.185
            -0.1 0.234 0.215
          ]
        }
        coordIndex [
          8, 7, 6, 5, 4, 3, 2, 1, 0, -1
          9, 10, 11, 12, 13, 14, 15, 16, 17, -1
          0, 1, 10, 9, -1, 1, 2, 11, 10, -1
          2, 3, 12, 11, -1, 3, 4, 13, 12, -1
          4, 5, 14, 13, -1, 5, 6, 15, 14, -1
          6, 7, 16, 15, -1, 7, 8, 17, 16, -1
        ]
      }
    }
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.636415 0.655556 0.655556
        }
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0 0.234 -0.174
            -0.056 0.234 -0.166
            -0.107 0.234 -0.145
            -0.155 0.234 -0.112
            -0.19 0.234 -0.064
            -0.19 0.234 0.074
            -0.16 0.234 0.096
            -0.16 0.234 0.151
            -0.155 0.234 0.2
            -0.107 0.234 0.236
            -0.056 0.234 0.256
            0 0.234 0.264
            0.056 0.234 0.256
            0.107 0.234 0.236
            0.155 0.234 0.2
            0.16 0.234 0.151
            0.16 0.234 0.096
            0.19 0.234 0.074
            0.19 0.234 -0.064
            0.155 0.234 -0.112
            0.107 0.234 -0.145
            0.056 0.234 -0.166
            0 0.234 -0.174
            0 0.245 -0.174
            -0.056 0.245 -0.166
            -0.107 0.245 -0.145
            -0.155 0.245 -0.112
            -0.19 0.245 -0.064
            -0.19 0.245 0.074
            -0.16 0.245 0.096
            -0.16 0.245 0.151
            -0.155 0.245 0.2
            -0.107 0.245 0.236
            -0.056 0.245 0.256
            0 0.245 0.264
            0.056 0.245 0.256
            0.107 0.245 0.236
            0.155 0.245 0.2
            0.16 0.245 0.151
            0.16 0.245 0.096
            0.19 0.245 0.074
            0.19 0.245 -0.064
            0.155 0.245 -0.112
            0.107 0.245 -0.145
            0.056 0.245 -0.166
            0 0.245 -0.174
          ]
        }
        coordIndex [
          22, 21, 20, 19, 18, 17, 16, 15, 14, 13
          12, 11, 10, 9, 8, 7, 6, 5, 4, 3
          2, 1, 0, -1, 23, 24, 25, 26, 27, 28
          29, 30, 31, 32, 33, 34, 35, 36, 37, 38
          39, 40, 41, 42, 43, 44, 45, -1, 0, 1
          24, 23, -1, 1, 2, 25, 24, -1, 2, 3
          26, 25, -1, 3, 4, 27, 26, -1, 4, 5
          28, 27, -1, 5, 6, 29, 28, -1, 6, 7
          30, 29, -1, 7, 8, 31, 30, -1, 8, 9
          32, 31, -1, 9, 10, 33, 32, -1, 10, 11
          34, 33, -1, 11, 12, 35, 34, -1, 12, 13
          36, 35, -1, 13, 14, 37, 36, -1, 14, 15
          38, 37, -1, 15, 16, 39, 38, -1, 16, 17
          40, 39, -1, 17, 18, 41, 40, -1, 18, 19
          42, 41, -1, 19, 20, 43, 42, -1, 20, 21
          44, 43, -1, 21, 22, 45, 44, -1
        ]
      }
    }
    Solid {
      translation -0.16038 0.082476 -6.71173e-05
      rotation 0.999999 0.00119113 -0.000120834 0.869288
      children [
        DEF WHEEL Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 0.0037037 0.0037037 0.0037037
                }
              }
              geometry Cylinder {
                height 0.037
                radius 0.0825
              }
            }
          ]
        }
      ]
      name "left wheel"
      boundingObject USE WHEEL
      physics Physics {
        density -1
        mass 0.05
      }
    }
    Solid {
      translation 0.160377 0.0824903 -6.57561e-05
      rotation 0.999999 0.000261099 0.00163841 0.569696
      children [
        USE WHEEL
      ]
      name "right wheel"
      boundingObject USE WHEEL
      physics Physics {
        density -1
        mass 0.05
      }
    }
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.136 0.184 -0.046
            0.136 0.184 -0.096
            0.101 0.184 -0.134
            0.054 0.184 -0.159
            0 0.184 -0.168
            -0.054 0.184 -0.159
            -0.101 0.184 -0.134
            -0.136 0.184 -0.096
            -0.136 0.184 -0.046
            0.136 0.184 -0.043
            0.136 0.234 -0.046
            0.136 0.234 -0.096
            0.101 0.234 -0.134
            0.054 0.234 -0.159
            0 0.234 -0.168
            -0.054 0.234 -0.159
            -0.101 0.234 -0.134
            -0.136 0.234 -0.096
            -0.136 0.234 -0.046
            0.136 0.234 -0.043
          ]
        }
        coordIndex [
          9, 8, 7, 6, 5, 4, 3, 2, 1, 0
          -1, 10, 11, 12, 13, 14, 15, 16, 17, 18
          19, -1, 0, 1, 11, 10, -1, 1, 2, 12
          11, -1, 2, 3, 13, 12, -1, 3, 4, 14
          13, -1, 4, 5, 15, 14, -1, 5, 6, 16
          15, -1, 6, 7, 17, 16, -1, 7, 8, 18
          17, -1, 8, 9, 19, 18, -1, 9, 0, 10
          19, -1
        ]
      }
    }
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.136 0.184 0.135
            -0.136 0.184 0.185
            -0.101 0.184 0.223
            -0.054 0.184 0.248
            0 0.184 0.257
            0.054 0.184 0.248
            0.101 0.184 0.223
            0.136 0.184 0.185
            0.136 0.184 0.135
            -0.136 0.184 0.135
            -0.136 0.234 0.135
            -0.136 0.234 0.185
            -0.101 0.234 0.223
            -0.054 0.234 0.248
            0 0.234 0.257
            0.054 0.234 0.248
            0.101 0.234 0.223
            0.136 0.234 0.185
            0.136 0.234 0.135
            -0.136 0.234 0.135
          ]
        }
        coordIndex [
          9, 8, 7, 6, 5, 4, 3, 2, 1, 0
          -1, 10, 11, 12, 13, 14, 15, 16, 17, 18
          19, -1, 0, 1, 11, 10, -1, 1, 2, 12
          11, -1, 2, 3, 13, 12, -1, 3, 4, 14
          13, -1, 4, 5, 15, 14, -1, 5, 6, 16
          15, -1, 6, 7, 17, 16, -1, 7, 8, 18
          17, -1, 8, 9, 19, 18, -1
        ]
      }
    }
    Transform {
      translation 0 0.0325 0.2147
      rotation 0 0 1 1.57
      children [
        Solid {
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 0 0 0
                }
              }
              geometry DEF BACK_WHEEL Cylinder {
                height 0.024
                radius 0.0325
              }
            }
          ]
          name "back wheel"
          boundingObject USE BACK_WHEEL
          physics Physics {
            density -1
            mass 0.05
          }
        }
      ]
    }
    DistanceSensor {
      translation -0.027 0.209 -0.164
      rotation 0 1 0 1.745
      children [
        DEF SONAR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 0.0517103
                }
              }
              geometry Cylinder {
                height 0.002
                radius 0.0175
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.077 0.209 -0.147
      rotation 0 1 0 2.094
      children [
        USE SONAR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.118 0.209 -0.116
      rotation 0 1 0 2.443
      children [
        USE SONAR
      ]
      name "ds2"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.136 0.209 -0.071
      rotation 0 1 0 3.14
      children [
        USE SONAR
      ]
      name "ds3"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.136 0.209 0.16
      rotation 0 1 0 -3.14
      children [
        USE SONAR
      ]
      name "ds4"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.118 0.209 0.205
      rotation 0 1 0 -2.443
      children [
        USE SONAR
      ]
      name "ds5"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.077 0.209 0.236
      rotation 0 1 0 -2.094
      children [
        USE SONAR
      ]
      name "ds6"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.027 0.209 0.253
      rotation 0 1 0 -1.745
      children [
        USE SONAR
      ]
      name "ds7"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.027 0.209 0.253
      rotation 0 1 0 -1.396
      children [
        USE SONAR
      ]
      name "ds8"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.077 0.209 0.236
      rotation 0 1 0 -1.047
      children [
        USE SONAR
      ]
      name "ds9"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.118 0.209 0.205
      rotation 0 1 0 -0.698
      children [
        USE SONAR
      ]
      name "ds10"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.136 0.209 0.16
      children [
        USE SONAR
      ]
      name "ds11"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.136 0.209 -0.071
      children [
        USE SONAR
      ]
      name "ds12"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.118 0.209 -0.116
      rotation 0 1 0 0.698
      children [
        USE SONAR
      ]
      name "ds13"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.077 0.209 -0.147
      rotation 0 1 0 1.047
      children [
        USE SONAR
      ]
      name "ds14"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.027 0.209 -0.164
      rotation 0 1 0 1.396
      children [
        USE SONAR
      ]
      name "ds15"
      lookupTable [
        0 1024 0
        0.5 0 0.1
      ]
      type "infra-red"
    }
  ]
  name "pioneer2"
  description "designed by Activmedia Robotics"
  contactMaterial "default0.0"
  boundingObject Transform {
    translation 0 0.13 0.0447
    children [
      Cylinder {
        height 0.24
        radius 0.219
      }
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "getdata"
  axleLength 0.32
  wheelRadius 0.0825
  speedUnit 0.1
}
